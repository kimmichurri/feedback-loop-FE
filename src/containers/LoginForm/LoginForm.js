import React, { Component } from 'react'

export class LoginForm extends Component {
  constructor() {
    super()
    this.state = {
        emailInput: '',
        passwordInput: '',
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({
      [name] : value
    })
  }

  checkUser = (e) => {
    e.preventDefault()
    this.props.handleLogin(this.state)
  }

  render() {
    return(
      <form className='login-form'>
        <div className='email'>
          <label>Email</label>
          <input 
            type="email" 
            onChange={this.handleChange}  
            name="emailInput"
            className="login-input"
          />
        </div>
        <div className='password'>
          <label>Password</label>
          <input 
            type="password" 
            onChange={this.handleChange}  
            name="passwordInput"
            className="login-input"
          />
        </div>
        <button className='login-button' onClick={this.checkUser}>
          Login
        </button>
        <button className='login-button' onClick={this.props.createAccount}>Create New Account</button>
      </form>
    )
  }
}

// export const mapDispatchToProps = (dispatch) => ({
//   setUser: (user) => dispatch(setUser(user))
// })

export default LoginForm