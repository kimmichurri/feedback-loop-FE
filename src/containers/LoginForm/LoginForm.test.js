import React from 'react'
import { LoginForm } from './LoginForm'
import { shallow } from 'enzyme'

describe('LoginForm', () => {
  let wrapper
  let mockFn = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<LoginForm 
      createAccount={mockFn}
      handleLogin={mockFn}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a default state', () => {
    const expected = {
      emailInput: '',
      passwordInput: '',
    }
    expect(wrapper.state()).toEqual(expected)
  })

  it('should update state on change', () => {
    const mockEvent = { target: { name: 'emailInput', value: 'user@gmail.com'}}
    wrapper.find('.email-input').simulate('change', mockEvent)
    expect(wrapper.state('emailInput')).toBe('user@gmail.com')
  })

  it('should call handleLogin when checkUser is called', () => {
    const preventDefault = { preventDefault: jest.fn() }
    wrapper.instance().checkUser(preventDefault)
    expect(mockFn).toHaveBeenCalled()
  })
})