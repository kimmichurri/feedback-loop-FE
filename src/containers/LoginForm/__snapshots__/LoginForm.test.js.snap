// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the snapshot 1`] = `
<form
  className="login-form"
>
  <div
    className="email"
  >
    <label>
      Email
    </label>
    <input
      className="login-input email-input"
      name="emailInput"
      onChange={[Function]}
      type="email"
    />
  </div>
  <div
    className="password"
  >
    <label>
      Password
    </label>
    <input
      className="login-input"
      name="passwordInput"
      onChange={[Function]}
      type="password"
    />
  </div>
  <button
    className="login-button"
    onClick={[Function]}
  >
    Login
  </button>
  <button
    className="login-button"
    onClick={[MockFunction]}
  >
    Create New Account
  </button>
</form>
`;
